cpp/modmesh/toggle/RadixTree.cpp

```diff
@@ -47,6 +47,30 @@ void CallProfiler::reset()
     m_cancel_callbacks.clear();
 }
 
+// Called when a function starts
+void CallProfiler::start_caller(const std::string & caller_name, const std::function<void()> & cancel_callback)
+{
+    m_cancel_callbacks.push_back(cancel_callback);
+    m_radix_tree.entry(caller_name);
+    CallerProfile & callProfile = m_radix_tree.get_current_node()->data();
+    callProfile.caller_name = caller_name;
+    callProfile.start_stopwatch();
+}
+
+// Called when a function ends
+void CallProfiler::end_caller()
+{
+    CallerProfile & call_profile = m_radix_tree.get_current_node()->data();
+    call_profile.stop_stopwatch(); // Update profiling information to the pending time and count
+    m_pending_nodes.insert(m_radix_tree.get_current_node());
+    m_radix_tree.move_current_to_parent(); // Pop the caller from the call stack
+
+    if (m_radix_tree.get_current_node() == m_radix_tree.get_root()) // If the root function ends, update all pending nodes and stable items
+    {
+        update_pending_nodes();
+        m_radix_tree.update_stable_items();
+    }
+}
```

Comment:
To address the re-entrancy issue, I added some additional members to the profiler. These members are updated only when the root function completes or when the profiler resets. As a result, if the profiler is serialized during profiling, the recorded data will reflect the state before the root function's execution was measured.

Reply:
Could you please summarize the variables you added for the reentrance support, and how it works, under this review comment?

Once it is clarified, we discuss whether or not the conclusion should be added in code comments.

Reply:
https://github.com/solvcon/modmesh/issues/320#issuecomment-2282744365 is an example for how to summarize the code mechanism.